bok_choy/web_app_test.py

```diff
@@ -56,6 +109,11 @@ def setUp(self):
         tags = [self.id()]
         self.browser = browser(tags, self.proxy)
 
+        # Needle uses these attributes for taking the screenshots
+        self.driver = self.get_web_driver()
+        self.driver.set_window_position(0, 0)
```

Comment:
Doh, that won't work. Screenshot assertions are a method of the WebAppTest object, and promises are related to PageObjects which WebAppTest objects don't know about.

Instead I'll update the example in the documentation to establish a pattern whereby you first assert that the element you want to capture the image of is visible, then do the screenshot assertion.